chapters/solutions.Rmd

```diff
@@ -1582,7 +1654,53 @@ most items would be checked off of the list.
 
 ### Exercise \@ref(packaging-ex-data) {-}
 
```

Comment:
Introducing new things like `__file__` and R data packages when the reader has only just seen (possibly for the first time) how to create a python package seems like it might be confusing.

I think we need an exercise before `packaging-ex-data` that does nothing more than ask the reader to go through the basic process of creating a package from scratch. For example:

### Exercise \@ref(packaging-ex-pratchett) {-}

Each chapter in this book opens with a quote from British author Terry Pratchett. Here's a script `quotes.py` which contains a function `random_quote` which prints a random Pratchett quote.

```
import random

quote_list = ["""It’s still magic even if you know how it’s done.""",
              "Ninety percent of most magic merely consists "\
              "of knowing one extra fact.",
              """There isn’t a way things should be. """\
              """There’s just what happens, and what we do.""",
              "Multiple exclamation marks are a sure sign "\
              "of a diseased mind.",
              "+++ Divide By Cucumber Error. """\
              "Please Reinstall Universe And Reboot +++",
              """It’s got three keyboards and a hundred extra knobs, """\
              """including twelve with ‘?’ on them.""",
              "Evil begins when you begin to treat people as things.",
             ]
                         
def random_quote():
    """Print a random Pratchett quote."""
    print(random.choice(quote_list))
```
Create a new (conda) development environment called `pratchett` and `pip` install a new package called `pratchett` into that environment. The package should contain `quotes.py` such that once the package has been `pip` installed the user could run:

```
from pratchett import quotes

quotes.random_quote()
``` 

### Solution

The directory tree for the `pratchett` package is:
```
pratchett
├── pratchett
│   └── quotes.py
├── README.md
└── setup.py
```

The `README.md` file should contain a basic description of the package and how to install/use it, while `setup.py` should read as follows:
```
from setuptools import setup


setup(
    name='pratchett',
    version='0.1',
    author='Amira Khan',
    packages=['pratchett'],
)
```
The following sequence of commands is required to create the development environment, activate that environment and then pip install the package:
```
$ conda create -n pratchett python
$ conda activate pratchett
(pratchett)$ cd pratchett
(pratchett)$ pip install -e .
```